set.mm

```diff
@@ -197624,6 +197650,44 @@ topology is based on the order and not the extended metric (which would
     VFACTVCPVEABUIQRVJVGVHOVCLZBMZVAVNUTABEUJUKVHVDVMVFBCOVCPVHCTSZVFBMVHVOOTSU
     LCOTUNUMCTABUOUPQRUQURUS $.
 
+  $( The value of a function with a domain of (at most) two elements.
+     (Contributed by Jim Kingdon, 25-Sep-2023.) $)
+  fvpr0o $p |- ( A e. V -> ( { <. (/) , A >. , <. 1o , B >. } ` (/) ) = A ) $=
+    ( c0 com wcel c1o wne cop cpr cfv wceq peano1 1n0 necomi fvpr1g mp3an13 ) D
+    EFACFDGHDDAIGBIJKALMGDNODGABECPQ $.
+
+  $( The value of a function with a domain of (at most) two elements.
+     (Contributed by Jim Kingdon, 25-Sep-2023.) $)
+  fvpr1o $p |- ( B e. V -> ( { <. (/) , A >. , <. 1o , B >. } ` 1o ) = B ) $=
+    ( c1o com wcel c0 wne cop cpr cfv wceq 1onn 1n0 necomi fvpr2g mp3an13 ) DEF
+    BCFGDHDGAIDBIJKBLMDGNOGDABECPQ $.
+
+  $( The value of a function with a domain of (at most) two elements.
+     (Contributed by Jim Kingdon, 25-Sep-2023.) $)
+  fvprif $p |- ( ( A e. V /\ B e. W /\ C e. 2o )
+      -> ( { <. (/) , A >. , <. 1o , B >. } ` C ) = if ( C = (/) , A , B ) ) $=
+    ( wcel c2o w3a c0 wceq cop c1o cpr cfv cif wa adantr simpr fveq2d 3eqtr4d
+    fvpr0o 3ad2ant1 iftrued fvpr1o 3ad2ant2 1n0 eqeq1d mtbiri iffalsed wo elpri
+    neii df2o3 eleq2s 3ad2ant3 mpjaodan ) ADFZBEFZCGFZHZCIJZCIAKLBKMZNZVAABOZJC
+    LJZUTVAPZIVBNZAVCVDUTVGAJZVAUQURVHUSABDUAUBQVFCIVBUTVARZSVFVAABVIUCTUTVEPZL
+    VBNZBVCVDUTVKBJZVEURUQVLUSABEUDUEQVJCLVBUTVERZSVJVAABVJVALIJLIUFULVJCLIVMUG
+    UHUITUSUQVAVEUJZURVNCILMGCILUKUMUNUOUP $.
```

Comment:
This looks a bit like `mptprop` (in my mathbox)!

In general, it would be nice if we could get a more general version of this theorem and maybe also `fnpr2ob`, using the more general ` { <. A , B >. , <. C , D >. } ` and `{ A , C }` where you have `2o`.
Then most of these follow with a simple substitution.

Reply:
I got partway down the path of doing this when I was making this branch, and I don't fully remember why I abandoned that. I might have been thinking of iset.mm (which would need decidable equality on V in mptprop) but of course iset.mm doesn't have to do exactly the same thing as set.mm, so that's not a definitive reason.